modules/system/database/migrations/2021_06_25_143510_Db_Attachment_ID_To_int.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+use Illuminate\Support\Facades\DB;
+use Winter\Storm\Database\Schema\Blueprint;
+use Winter\Storm\Database\Updates\Migration;
+use Winter\Storm\Support\Facades\Schema;
+
+class DbAttachmentIdToInt extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('system_files', function (Blueprint $table) {
+            // Drop old indexes
+            $table->dropIndex(['attachment_id']);
+            $table->dropIndex(['attachment_type']);
+            $table->dropIndex(['field']);
+
+            $table->index([
+                'field',
+                'attachment_type',
+                'attachment_id',
+            ]);
+        });
+
+        // Convert attachment_id to bigint if the attachment_id column only contains integer values.
+        $convert = true;
+        DB::table('system_files')
+            ->select(['id', 'attachment_id'])
+            ->chunkById(1000, function ($attachmentIds) use (&$convert) {
+                $files = $attachmentIds->filter(function ($file) {
+                    return ! is_numeric($file->attachment_id);
```

Comment:
@ericp-mrel this might be a more reliable way of determining if an attachment ID is an integer. A conversion of the attachment ID to an integer then to a string should result in the same value as a conversion straight to a string if the ID truly is an integer. It would also help resolve @LukeTowers' correct comment about some numerics passing this check even though they are not an integer.

```suggestion
                    return ((string) $file->attachment_id === (string) intval($file->attachment_id));
```